calico_containers/pycalico/datastore.py

```diff
@@ -137,17 +139,55 @@ class DatastoreClient(object):
     """
 
     def __init__(self):
+        etcd_endpoints = os.getenv(ETCD_ENDPOINTS_ENV, ETCD_ENDPOINTS_DEFAULT)
         etcd_authority = os.getenv(ETCD_AUTHORITY_ENV, ETCD_AUTHORITY_DEFAULT)
-        if not validate_hostname_port(etcd_authority):
-            raise DataStoreError("Invalid %s. It must take the form "
-                                 "<address>:<port>. Value provided is '%s'" %
-                                 (ETCD_AUTHORITY_ENV, etcd_authority))
-
-        (host, port) = etcd_authority.split(":", 1)
         etcd_scheme = os.getenv(ETCD_SCHEME_ENV, ETCD_SCHEME_DEFAULT)
         etcd_key = os.getenv(ETCD_KEY_FILE_ENV, '')
         etcd_cert = os.getenv(ETCD_CERT_FILE_ENV, '')
         etcd_ca = os.getenv(ETCD_CA_CERT_FILE_ENV, '')
+
+        etcd_addrs_str = []
+        if etcd_endpoints != ETCD_ENDPOINTS_DEFAULT:
+            # ETCD_ENDPOINTS specified: use it to determine scheme and etcd
+            # location.
+            endpoints = [x.strip() for x in etcd_endpoints.split(",")]
+            try:
+                scheme = None
+                for e in endpoints:
+                    s, a = e.split("://")
+                    etcd_addrs_str.append(a)
+                    if scheme == None:
```

Comment:
Why not just set etcd_scheme = s and drop this if test and the assignment below?